### DISCUSSION TITLE
Bundled YAML grammar duplicates VS Code built-in grammar

### DISCUSSION THREAD

**Discussion by prettyv** [redhat-developer/vscode-yaml#831] [General]
In the Developer Console I noticed a warning pertaining to the shipped TextMate grammar:

```
TMScopeRegistry.ts:47 Overwriting grammar scope name to file mapping for scope source.yaml.
Old grammar file: file:///usr/lib/code/extensions/yaml/syntaxes/yaml.tmLanguage.json.
New grammar file: file:///home/prettyv/.vscode-oss/extensions/redhat.vscode-yaml-1.10.1/syntaxes/yaml.tmLanguage.json
```

That means VS Code already ships a grammar for YAML that is then overwritten by the one provided in this extension on startup. I compared the grammar shipped by VS Code and the one provided here and the only difference is (I suspect unnecessary) escaping of forward slashes in one line:
```diff
@@ -249,5 +249,5 @@
                }
            },
-           "match": "((\\*))([^\\s\\[\\]/{/},]+)([^\\s\\]},]\\S*)?"
+           "match": "((\\*))([^\\s\\[\\]\/{\/},]+)([^\\s\\]},]\\S*)?"
        },
        "flow-collection": {
```

This leads me to believe that the grammar could actually be dropped from the extension. Are there variants of VS Code where it is not guaranteed that the grammar will be built-in and having it in the extension would then help?

**Comment by gorkem** [comment]
The extension is also used in Theia, I do not remember exactly, but possibly that was the reason for including the grammar.
